src/bgw/job.c

```diff
@@ -81,10 +82,12 @@ bgw_job_from_tuple(HeapTuple tuple, size_t alloc_size, MemoryContext mctx)
 {
 	BgwJob	   *job;
 
-	/* allow for embedding */
+	/*
+	 * allow for embedding with arbitrary alloc_size, which means we can't use
+	 * the STRUCT_FROM_TUPLE macro
+	 */
 	Assert(alloc_size >= sizeof(BgwJob));
-	job = MemoryContextAllocZero(mctx, alloc_size);
-	memcpy(&job->fd, GETSTRUCT(tuple), sizeof(FormData_bgw_job));
+	job = (BgwJob *) create_struct_from_tuple(tuple, mctx, alloc_size, sizeof(FormData_bgw_job));
```

Comment:
Is this actually a correct change? Before we were `memcpy`ing to `&job->fd`, but now it appears we are `memcpy`ing to just `job`?

Reply:
I guess that seems reasonable to me, not super familiar with compiler magic.

Reply:
It's one of those weird things that's kinda in the standard but not explicit: a pointer to a struct is always a pointer to the first member (when "suitably converted"), and this is one of those things which is commonly assumed must work in C, and we have some compiler devs saying they'll keep it valid. If this stops working a good bit of postgres will break, so I don't think we need to worry about it ourselves.